src/api/procedures/setTransferRestrictions.ts

```diff
@@ -147,31 +195,40 @@ export async function prepareSetTransferRestrictions(
   }
 
   const transactions = [];
-
-  const op =
-    type === TransferRestrictionType.Count
-      ? statisticsOpTypeToStatOpType(StatisticsOpType.Count, context)
-      : statisticsOpTypeToStatOpType(StatisticsOpType.Balance, context);
+  const op = transferRestrictionTypeToStatOpType(type, context);
   transactions.push(
     checkTxType({
       transaction: statistics.setAssetTransferCompliance,
       args: [tickerKey, complianceConditionsToBtreeSet(conditions, context)],
     })
   );
 
-  if (exemptions.length) {
-    const exemptedIds = await getExemptedIds(exemptions, context, ticker);
-    const exemptedScopeIds = exemptedIds.map(entityId => stringToIdentityId(entityId, context));
-    const btreeIds = scopeIdsToBtreeSetIdentityId(exemptedScopeIds, context);
-    const exemptKey = toExemptKey(tickerKey, op);
+  Object.entries(toSetExemptions).forEach(([claimType, exempted]) => {
+    const rawClaimType = claimType === 'None' ? undefined : (claimType as ClaimType);
+    const exemptKey = toExemptKey(tickerKey, op, rawClaimType);
+    const exemptedBtreeSet = identitiesToBtreeSet(exempted, context);
     transactions.push(
       checkTxType({
         transaction: statistics.setEntitiesExempt,
-        feeMultiplier: new BigNumber(exemptions.length),
-        args: [true, exemptKey, btreeIds],
+        feeMultiplier: new BigNumber(exemptedBtreeSet.size),
+        args: [true, exemptKey, exemptedBtreeSet],
       })
     );
-  }
+  });
+
+  Object.entries(toRemoveExemptions).forEach(([claimType, unExempted]) => {
```

Comment:
lodash `forEach`

Reply:
Also, these two blocks look very similar, you could probably refactor them into a named arrow function